# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmConstants.h
# Commit: 72bf9c871ded
# Full Hash: 72bf9c871ded006f56f09b24d0723365516445bb
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 5
# Description:
#   Bug 1692065 - wasm: Add tagging for 31-bit integers and i31ref. r=yury
#   
#   This commit adds a tag 0x1 for 31-bit integers in AnyRef. Any JS value
#   that is losslessly representable (without coercion) in this range is
#   stored without boxing. This is chosen to line up exactly with how the
# ==============================================================================

diff -r 28feda14fdec -r 72bf9c871ded js/src/wasm/WasmConstants.h
--- a/js/src/wasm/WasmConstants.h	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/wasm/WasmConstants.h	Fri Aug 04 14:06:50 2023 +0000
@@ -84,6 +84,9 @@
   // Type constructor for non-nullable reference types.
   Ref = 0x6b,  // SLEB128(-0x15)
 
+  // A reference to an unboxed 31-bit integer.
+  I31Ref = 0x6a,  // SLEB128(-0x16)
+
   // A null reference in the extern hierarchy.
   NullExternRef = 0x69,  // SLEB128(-0x17)
 
@@ -505,6 +508,11 @@
   ArrayCopy = 0x18,
   ArrayLen = 0x19,
 
+  // I31 operations
+  I31New = 0x20,
+  I31GetS = 0x21,
+  I31GetU = 0x22,
+
   // Ref operations
   RefTestV5 = 0x44,
   RefCastV5 = 0x45,